File: fyn-flutter-app/lib/features/auth/data/models/auth_response.dart
import 'package:json_annotation/json_annotation.dart';
import 'user_response.dart';

part 'auth_response.g.dart';

@JsonSerializable()
class AuthResponse {
  final String accessToken;
  final String refreshToken;
  final int expiresIn;
  final UserResponse user;

  AuthResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.expiresIn,
    required this.user,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) =>
      _$AuthResponseFromJson(json);

  Map<String, dynamic> toJson() => _$AuthResponseToJson(this);
}

File: fyn-monolithic/src/main/java/com/fyn_monolithic/dto/response/auth/AuthResponse.java
package com.fyn_monolithic.dto.response.auth;

import com.fyn_monolithic.dto.response.user.UserResponse;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
public class AuthResponse {
    String accessToken;
    String refreshToken;
    long expiresIn;
    UserResponse user;
}
